manim/scene/scene.py

```diff
@@ -849,8 +883,36 @@ def get_time_progression(
         if self.renderer.skip_animations and not override_skip_animations:
             times = [run_time]
         else:
-            step = 1 / config["frame_rate"]
-            times = np.arange(0, run_time, step)
+            if speedinfo is not None:
+                nodes = list(speedinfo.keys())
+                speeds = list(speedinfo.values())
+                if nodes[0] != 0:
+                    nodes.insert(0, 0)
+                if nodes[-1] != 1:
+                    nodes.append(1)
+                    speeds.append(speeds[-1])
+                times = np.array([])
+                on_node = 0
+
+                for dur, s in zip(np.diff(nodes), speeds):
+                    rel_speed_change = s / self.speed
+                    adj_func = np.vectorize(
+                        lambda x: (rel_speed_change ** 2 - 1) / 4 * x ** 2 + x,
```

Comment:
What does this function do, precisely? What if users want to use a custom function here?